catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -109,6 +118,45 @@ public ContentDirectoryMonitor(CamelContext camelContext, int maxRetries,
     }
 
     /**
+     * Set the thread pool size with the given argument.  If the given argument is less than 1,
+     * numThreads is set to 1.  If the given argument is greater than the MAX_THREAD_SIZE, numThreads
+     * is set to MAX_THREAD_SIZE.
+     *
+     * @param numThreads - the specified size to make the thread pool
+     */
+    public void setNumThreads(Integer numThreads) {
+        if (numThreads < 1) {
```

Comment:
Instead of the if-statements, how about:

this.numThreads = Math.max(numThreads, MIN_THREAD_SIZE);
this.numThreads = Math.min(this.numThreads, MAX_THREAD_SIZE);

That will reduce the branch-complexity.